src/core.rs

```diff
@@ -61,6 +64,9 @@ const MAX_JOINING_NODES: usize = 1;
 /// of joining nodes.
 const JOINING_NODE_TIMEOUT_SECS: i64 = 5;
 
+/// Time (in seconds) after which bootstrap is cancelled (and possibly retried).
+const BOOTSTRAP_TIMEOUT_SECS: i64 = 10;
```

Comment:
The timeout might be again too optimistic, but it's easy to change later when necessary.